ui/src/app/applications/components/applications-terminate-sync-panel/applications-terminate-sync-panel.tsx

```diff
@@ -0,0 +1,118 @@
+import {ErrorNotification, NotificationType, SlidingPanel} from 'argo-ui';
+import * as React from 'react';
+import {Form, FormApi} from 'react-form';
+import {ProgressPopup, Spinner} from '../../../shared/components';
+import {Consumer, ContextApis} from '../../../shared/context';
+import * as models from '../../../shared/models';
+import {services} from '../../../shared/services';
+import {ApplicationSelector} from '../../../shared/components';
+import {confirmSyncingAppOfApps} from '../utils';
+
+interface Progress {
+    percentage: number;
+    title: string;
+}
+
+export const ApplicationsTerminateSyncPanel = ({show, apps, hide}: {show: boolean; apps: models.Application[]; hide: () => void}) => {
+    const [form, setForm] = React.useState<FormApi>(null);
+    const [progress, setProgress] = React.useState<Progress>(null);
+    const getSelectedApps = (params: any) => apps.filter((_, i) => params['app/' + i]);
+    const [isPending, setPending] = React.useState(false);
+    const syncHandler = (currentForm: FormApi, ctx: ContextApis, applications: models.Application[]) => {
+        const formValues = currentForm.getFormState().values;
+        const replaceChecked = formValues.syncOptions?.includes('Replace=true');
+        const selectedApps = [];
+        const selectedAppOfApps: models.Application[] = [];
+        let containAppOfApps = false;
+
+        for (const key in formValues) {
+            if (key.startsWith('app/') && formValues[key]) {
+                selectedApps.push(applications[parseInt(key.slice(key.lastIndexOf('/') + 1), 10)]);
+            }
+        }
+
+        selectedApps.forEach(app => {
+            if (app.isAppOfAppsPattern) {
+                containAppOfApps = true;
+                selectedAppOfApps.push(app);
+            }
+        });
+
+        if (replaceChecked && containAppOfApps) {
+            confirmSyncingAppOfApps(selectedAppOfApps, ctx, currentForm).then(confirmed => {
```

Comment:
The naming is a bit confusing. `ApplicationsTerminateSyncPanel` suggests it's only for termination of sync, so why do we confirm sync and check replace option? If we need to, shall we check prune as well?

Reply:
fixed, extract common logic and remove sync related code.